src/test/java/com/gotocompany/firehose/serializer/TypecastedJsonSerializerTest.java

```diff
@@ -0,0 +1,121 @@
+
+    @Before
+    public void setup() {
+        messageSerializer = Mockito.mock(MessageSerializer.class);
+        serializerConfig = Mockito.mock(SerializerConfig.class);
+        Map<String, Function<String, Object>> property = serializerConfigConverter.convert(null, DEFAULT_PARAMETERS);
+        Mockito.when(messageSerializer.serialize(Mockito.any())).thenReturn(DEFAULT_JSON_MESSAGE);
+        Mockito.when(serializerConfig.getJsonTypecastMapping()).thenReturn(property);
+        typecastedJsonSerializer = new TypecastedJsonSerializer(
+                messageSerializer, serializerConfig
+        );
+    }
+
+    @Test
+    public void serializeShouldCastToNumberWhenGivenMessageWithQuoteWrappedNumberAndMatchingJsonPathConfiguration() {
+        String processedJsonString = typecastedJsonSerializer.serialize(buildMessage("key", DEFAULT_JSON_MESSAGE));
+        DocumentContext jsonPath = JsonPath.parse(processedJsonString);
+        JSONArray integerJsonArray = jsonPath.read("$..int");
+        JSONArray longJsonArray = jsonPath.read("$..long");
+        JSONArray doubleJsonArray = jsonPath.read("$..double");
+
+        Assertions.assertTrue(integerJsonArray.get(0) instanceof Integer);
+        Assertions.assertTrue(longJsonArray.get(0) instanceof Long);
+        Assertions.assertTrue(doubleJsonArray.get(0) instanceof Double);
+        Assertions.assertEquals(integerJsonArray.get(0), 1234);
+        Assertions.assertEquals(longJsonArray.get(0), 1234568129012312L);
+        Assertions.assertEquals(doubleJsonArray.get(0), 12.1);
+    }
+
+    @Test
+    public void serializeShouldIgnoreWhenGivenNullMessageValue() {
+        String jsonWithNullMappedValue = "{\"key\": \"value\", \"long\":null}";
+        Mockito.when(messageSerializer.serialize(Mockito.any())).thenReturn(jsonWithNullMappedValue);
+        String processedJsonString = typecastedJsonSerializer.serialize(buildMessage("key", jsonWithNullMappedValue));
+        DocumentContext jsonPath = JsonPath.parse(processedJsonString);
+        JSONArray fieldWithValue = jsonPath.read("$..key");
+        JSONArray integerJsonArray = jsonPath.read("$..long");
+
+        Assertions.assertEquals("value", fieldWithValue.get(0));
+        Assertions.assertNull(integerJsonArray.get(0));
+    }
+
+    @Test
+    public void serializeShouldReturnMessageAsItIsWhenNoJsonPathConfigurationGiven() {
+        Mockito.when(serializerConfig.getJsonTypecastMapping()).thenReturn(new HashMap<>());
+        typecastedJsonSerializer = new TypecastedJsonSerializer(
+                messageSerializer, serializerConfig
+        );
+
+        String result = typecastedJsonSerializer.serialize(buildMessage("key", DEFAULT_JSON_MESSAGE));
+
+        Assertions.assertEquals(JsonPath.parse(DEFAULT_JSON_MESSAGE).jsonString(), JsonPath.parse(result).jsonString());
+    }
+
+    @Test
+    public void serializeShouldReturnMessageAsItIsWhenJsonPathConfigurationDoesNotMatch() {
```

Comment:
remove serialize from the starting of the test name , test name should with "should"